api/src/main/java/org/pmiops/workbench/iam/IamServiceImpl.java

```diff
@@ -72,40 +84,42 @@ private void grantServiceAccountUserRole(String googleProject, String petService
   }
 
   @Override
-  public void grantWorkflowRunnerRoleForUsers(String googleProject, List<String> userEmails) {
-    GoogleApi googleApiAsImpersonatedUser = new GoogleApi();
-    try {
-      List<String> petServiceAccountsToGrantPermission = new ArrayList<>();
-      for (String userEmail : userEmails) {
-        googleApiAsImpersonatedUser.setApiClient(
-            samApiClientFactory.newImpersonatedApiClient(userEmail));
+  public List<String> grantWorkflowRunnerRoleForUsers(
+      String googleProject, List<String> userEmails) {
+    List<String> petServiceAccountsToGrantPermission = new ArrayList<>();
+    List<String> petServiceAccountFailures = new ArrayList<>();
+
+    for (String userEmail : userEmails) {
+      try {
         String petServiceAccountName =
-            getOrCreatePetServiceAccount(googleProject, googleApiAsImpersonatedUser);
+            getOrCreatePetServiceAccountUsingImpersonation(googleProject, userEmail);
         petServiceAccountsToGrantPermission.add(petServiceAccountName);
         grantServiceAccountUserRole(googleProject, petServiceAccountName);
+      } catch (Exception e) {
+        petServiceAccountFailures.add(userEmail);
```

Comment:
key difference: collects individual email failures but keeps going to try others